tools/sgen/input/FluentUIStyle.yml

```diff
@@ -190,28 +190,66 @@ AP_MSFAvatarTokens:
-                                    "NamedColor(gray30)",
-                                    "NamedColor(blueMagenta30)",
-                                    "NamedColor(gray20)",
-                                    "NamedColor(cyan30)",
-                                    "NamedColor(orange30)" ]
-  textColor: $Colors.Foreground.neutralInverted
+  textCalculatedBackgroundColors: [ "FluentUIColor(light: #D69BA5, dark: #420610)",
+                                    "FluentUIColor(light: #EDACB1, dark: #6E0911)",
+                                    "FluentUIColor(light: #F1BBBD, dark: #751D20)",
+                                    "FluentUIColor(light: #EFC4AD, dark: #712D09)",
+                                    "FluentUIColor(light: #FFDDB3, dark: #8F4F00)",
+                                    "FluentUIColor(light: #F9E2AE, dark: #835C00)",
+                                    "FluentUIColor(light: #EDDEA6, dark: #6D5700)",
+                                    "FluentUIColor(light: #E0CFA2, dark: #563F06)",
+                                    "FluentUIColor(light: #DDC3B0, dark: #50301A)",
+                                    "FluentUIColor(light: #BDDA9B, dark: #294903)",
+                                    "FluentUIColor(light: #A8F0CD, dark: #00723B)",
+                                    "FluentUIColor(light: #9AD39A, dark: #063C06)",
+                                    "FluentUIColor(light: #A6E8ED, dark: #00656D)",
+                                    "FluentUIColor(light: #9BD9DB, dark: #02494C)",
+                                    "FluentUIColor(light: #95C8D4, dark: #00333F)",
+                                    "FluentUIColor(light: #A9D3F2, dark: #004377)",
+                                    "FluentUIColor(light: #9ABFDD, dark: #002B4F)",
+                                    "FluentUIColor(light: #C7D1FA, dark: #2C3C85)",
+                                    "FluentUIColor(light: #A3B2E9, dark: #001665)",
+                                    "FluentUIColor(light: #D2CCF8, dark: #403582)",
+                                    "FluentUIColor(light: #C6B1DE, dark: #341A51)",
+                                    "FluentUIColor(light: #DAA7E0, dark: #4D0D56)",
+                                    "FluentUIColor(light: #E7BFED, dark: #63276D)",
+                                    "FluentUIColor(light: #F7C0E3, dark: #7F215D)",
+                                    "FluentUIColor(light: #ECA5D1, dark: #6B0042)",
+                                    "FluentUIColor(light: #D696C0, dark: #43002C)",
+                                    "FluentUIColor(light: #D7D5D4, dark: #454241)",
+                                    "FluentUIColor(light: #CECCCB, dark: #333231)",
+                                    "FluentUIColor(light: #CDD5D8, dark: #3A4346)",
+                                    "FluentUIColor(light: #BCC3C7, dark: #1F2427)" ]
+  textCalculatedForegroundColors: [ "FluentUIColor(light: #420610, dark: #D69BA5)",
```

Comment:
how do we make sure the text foreground color array size matches the avatar background colors size?

Reply:
They are in the same order and are returned by the same hash function.

Reply:
these colors don't have fluent palette name?

Reply:
As we spoke offline, I found the list below but they were not defined on a palette on that figma file. Following up with design to get the Vnext version of the palette.

Dark Red
Cranberry
Red
Pumpkin
Peach
Marigold
Gold
Brass
Brown
Forest
Seafoam
Dark Green
Light Teal
Teal
Steel
Blue
Royal Blue
Cornflower
Navy
Lavendar
Purple
Grape
Lilac
Pink
Magenta
Plum
Beige
Mink
Platinum
Anchor

Reply:
Just updated with these colors created from the shared palette.

Reply:
Added unit tests to validate that the number of the colors defined for both tokens are the same.
Also testing that they match (foreground dark should match background light and background dark should match foreground light).